.github/workflows/pythonbuild.yml

```diff
@@ -13,35 +13,22 @@ jobs:
     strategy:
       matrix:
         include:
-          - python-version: "3.7"
-            tox-env: py37-core
-          - python-version: "3.8"
-            tox-env: py38-core
-          - python-version: "3.9"
-            tox-env: py39-core
-          - python-version: "3.10"
-            tox-env: py310-core
-          - python-version: "3.11"
-            tox-env: py311-core
-          - python-version: "3.12"
-            tox-env: py312-core
+          - tox-env: py38-core
+          - tox-env: py39-core
+          - tox-env: py310-core
+          - tox-env: py311-core
+          - tox-env: py312-core
 
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
+      - name: Set up the latest version of uv
+        uses: astral-sh/setup-uv@v5
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Cache pip
-        uses: actions/cache@v4
-        with:
-          # This path is specific to Ubuntu
-          path: ~/.cache/pip
-          # Look to see if there is a cache hit for the corresponding requirements files
-          key: ${{ format('{0}-pip-{1}', runner.os, hashFiles('dev-requirements.txt', format('requirements{0}.txt', matrix.spark-version-suffix))) }}
+          enable-cache: true
+          cache-dependency-glob: "pyproject.toml"
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip 'tox<4.0'
+          uv tool install --python-preference only-managed --python 3.12 tox --with tox-uv
```

Comment:
I'm a bit confused how the CI is working know to use the correct python version for each tox environment. Could you help me understand why we use py 3.12 to install tox, but expect the correct python version to run the specific tox environment tests? Are there details within `tox-uv` that handle this?

Reply:
> Are there details within tox-uv that handle this?
Yes, you are right.
ref: https://github.com/tox-dev/tox-uv

if you run `uvx --with tox-uv tox -e py311` then we can see the log below:
```shell
$ uvx --with tox-uv tox -e py311
py311: uv-sync> uv sync --frozen --python-preference system --no-dev -p cpython3.11
...
```

At the first line of the log shows constructing process of Python 3.11 environment.

Reply:
Thanks for the confirmation